src/fio.c

```diff
@@ -339,3 +341,146 @@ b32 fio_write_flash(s32 pageNum, s8* readBuffer, u32 numBytes) {
 void fio_erase_flash(s32 pageNum) {
     osFlashSectorErase(pageNum * sizeof(SaveGlobals));
 }
+
+void fio_copy_s8_array(s8* dest, s8* src, s32 srcMax, s32 destMax) {
+    s32 i;
+    for (i = 0; i < srcMax; i++) {
+        dest[i] = src[i];
+    }
+    for (; i < destMax; i++) {
+        dest[i] = 0;
+    }
+}
+
+void fio_copy_s16_array(s16* dest, s16* src, s32 srcMax, s32 destMax) {
+    s32 i;
+    for (i = 0; i < srcMax; i++) {
+        dest[i] = src[i];
+    }
+    for (; i < destMax; i++) {
+        dest[i] = 0;
+    }
+}
+
+void fio_copy_s32_array(s32* dest, s32* src, s32 srcMax, s32 destMax) {
+    s32 i;
+    for (i = 0; i < srcMax; i++) {
+        dest[i] = src[i];
+    }
+    for (; i < destMax; i++) {
+        dest[i] = 0;
+    }
+}
+
+#define COPY_S8_ARRAY(new, old)   fio_copy_s8_array(new, old, ARRAY_COUNT(old), ARRAY_COUNT(new));
+#define COPY_S16_ARRAY(new, old) fio_copy_s16_array(new, old, ARRAY_COUNT(old), ARRAY_COUNT(new));
+#define COPY_S32_ARRAY(new, old) fio_copy_s32_array(new, old, ARRAY_COUNT(old), ARRAY_COUNT(new));
+
+void deserialize_vanilla_save(SaveData* newSave) {
+    VanillaSaveData oldSave;
+    s32 i, j;
+
+    memcpy(&oldSave, newSave, sizeof(VanillaSaveData));
+    memset(newSave, 0, sizeof(SaveData));
+
+    strcpy(newSave->magicString, oldSave.magicString);
+    strcpy(newSave->version, DX_MOD_VERSION_STRING); //TODO ensure 31 chars max
```

Comment:
config.h says that you can comment out DX_MOD_VERSION_STRING, but this will make this not compile. We should either remove that note in config.h, or use a fallback (e.g. "dx") if its not defined. Also config.h should say that that string is used for versioning save files